pkg/catalog/catalog.go

```diff
@@ -23,9 +23,9 @@ import (
 	"github.com/hashicorp/go-multierror"
 	lru "github.com/hnlq715/golang-lru"
 	"github.com/rs/xid"
+	"github.com/treeverse/lakefs/modules/block/factory"
```

Comment:
The run file already imports and instantiate the block adapter, however catalog also instantiates it's own block adapter, this is probably for historical reasons.
Consider - passing the block adapter in run to the catalog `New` function

Reply:
If I get what you're suggesting - 
In the `run` file, the `BuildBlockAdapter` is initialized with statsCollector, while in the `catalog` it isn't, no?